presto-main/src/main/java/com/facebook/presto/ForkJoinThreadFactoryBuilder.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto;
+
+import java.util.concurrent.ForkJoinPool;
+import java.util.concurrent.ForkJoinPool.ForkJoinWorkerThreadFactory;
+import java.util.concurrent.ForkJoinWorkerThread;
+import java.util.concurrent.atomic.AtomicLong;
+
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class ForkJoinThreadFactoryBuilder
```

Comment:
The reason why we need a specific `ForkJoinThreadFactoryBuilder` (instead of using  what JDK provides) is because of want to have specific thread name and/or context class loader, right ?

Reply:
> Any idea why JDK just don't provided a bounded + cached thread pool ? :)

That's a very good question. I guess the ForkJoinPool is some way of providing it. `BoundedExecutor` from the Airlift is yet another solution, but unfortunately it is very limited. It is an `Executor` and not an executor service.

Reply:
> As a future work, does it make sense to integrate/refactor ForkJoinThreadFactoryBuilder into airlift? -- And replace BoundedExecutor + cached thread pool pattern when appropriate ?

I think yes, the default choice should be the `ForkJoinPool`

Reply:
> The reason why we need a specific ForkJoinThreadFactoryBuilder (instead of using what JDK provides) is because of want to have specific thread name and/or context class loader, right ?

Yeah, that's the only reason. Just a convenience builder.